input.c

```diff
@@ -640,7 +671,9 @@ void ip_setup(struct input_plugin *ip)
 int ip_close(struct input_plugin *ip)
 {
 	int rc;
-
+	struct connection *conn = &ip->data.conn;
+	if (conn->ssl != NULL)
+		ssl_close(conn->ssl);
```

Comment:
`connection->close()` which would then internally check to see if `fd` and `ssl` are set.